fenzo-core/src/main/java/com/netflix/fenzo/queues/tiered/ResAllocsUtil.java

```diff
@@ -0,0 +1,92 @@
+package com.netflix.fenzo.queues.tiered;
+
+import com.netflix.fenzo.queues.QueuableTask;
+import com.netflix.fenzo.sla.ResAllocs;
+import com.netflix.fenzo.sla.ResAllocsBuilder;
+
+import java.util.List;
+
+/**
+ * Collection of helper functions for {@link com.netflix.fenzo.sla.ResAllocs} data type.
+ */
+public final class ResAllocsUtil {
+
+    public static ResAllocs addAll(List<ResAllocs> resToAdd) {
+        if (resToAdd.isEmpty()) {
+            throw new IllegalArgumentException("Empty list provided");
+        }
+        if (resToAdd.size() == 1) {
+            return resToAdd.get(0);
+        }
+        double cores = 0;
+        double memory = 0;
+        double network = 0;
+        double disk = 0;
+        for (ResAllocs r : resToAdd) {
+            cores += r.getCores();
+            memory += r.getMemory();
+            network += r.getNetworkMbps();
+            disk += r.getDisk();
+        }
+        return new ResAllocsBuilder(resToAdd.get(0).getTaskGroupName())
+                .withCores(cores)
+                .withMemory(memory)
+                .withNetworkMbps(network)
+                .withDisk(disk)
+                .build();
+    }
+
+    public static ResAllocs subtract(ResAllocs first, ResAllocs second) {
+        return new ResAllocsBuilder(first.getTaskGroupName())
+                .withCores(first.getCores() - second.getCores())
```

Comment:
My assumption is that negative values are ok, and there are many use cases for it.
In this PR it is ok to get result with negative values, which would indicate that we over-provisioned resources. This may happen if we scale down tier or add new capacity groups.